src/basic/uid-classification.h

```diff
@@ -33,4 +14,44 @@ typedef struct UGIDAllocationRange {
 int read_login_defs(UGIDAllocationRange *ret_defs, const char *path, const char *root);
 const UGIDAllocationRange *acquire_ugid_allocation_range(void);
 
+typedef enum UGIDRangeFlags {
+        UGID_RANGE_SYSTEM     = 1 << 0,
+        UGID_RANGE_DYNAMIC    = 1 << 1,
+        UGID_RANGE_CONTAINER  = 1 << 2,
+} UGIDRangeFlags;
+
+static inline bool uid_in_range(uid_t uid, UGIDRangeFlags flags) {
+        assert(flags);
+
+        if (FLAGS_SET(flags, UGID_RANGE_DYNAMIC))
+                if (DYNAMIC_UID_MIN <= uid && uid <= DYNAMIC_UID_MAX)
+                        return true;
+
+        if (FLAGS_SET(flags, UGID_RANGE_CONTAINER))
+                if (CONTAINER_UID_BASE_MIN <= uid && uid <= CONTAINER_UID_BASE_MAX)
+                        return true;
+
+        if (FLAGS_SET(flags, UGID_RANGE_SYSTEM)) {
+                const UGIDAllocationRange *defs;
+                assert_se(defs = acquire_ugid_allocation_range());
+
+                if (uid <= defs->system_uid_max)
+                        return true;
+        }
+
+        return false;
+}
```

Comment:
I am not convinced this is really an improvement over the status quo, tbh. The functions were very easy to grok before and simple, but they aren't really anymore, they are a lot more opaque.

I am not sure what is won by making this flags-based tbh.

Also, such long functions should not be inline.

Reply:
As explained in the commit message, I find the `uid_in_range()` name more explicit than `uid_is_*()` ones. The former clearly indicates that the test is done against the static UID ranges (only) and the test is only relevant during allocations of UIDs and not for checking that an UID has been assigned to a dynamic user, container...

Reply:
`uid_in_range()` and `gid_in_range()` were made inline because I assumed that the compiler would simplify them drastically  given that `flags` argument is always a constant.

Reply:
we use LTO, which allows the compiler to inline code across object files if it wants. The compiler should make its choice really on its own.